### PR TITLE
[BE-1313] Commit usermapping

### PR DESCRIPTION
**Pull request by Robindiddams**
**Description:**

now each git integration emits a `user_mapping.json.gz` for later mapping

### CODE REVIEW

**Comment by Robindiddams** [comment]
gotta fix the merge and also test ingesting with pipeline

**Comment by keegandonley** [comment]
Ah gotta love the commit message emojis @Robindiddams

**Comment by Robindiddams** [comment]
@keegandonley I bring color to software development

File: integrations/bitbucket/export.go [hunk 1]
```diff
@@ -42,6 +46,101 @@ var gitIntegration = git.Integration{
...
+						Avatar struct {
+							Href string `json:"href"`
+						} `json:"avatar"`
+					} `json:"links"`
+				} `json:"user"`
+			} `json:"author"`
+		}
+		userDuplicateMap := make(map[string]bool)
+		for _, email := range git.IgnoredEmails {
+			userDuplicateMap[hash.Values(email)] = true
```
Comment (jhaynie) [inline integrations/bitbucket/export.go]: seed the hash with customerID

File: integrations/bitbucket/export.go [hunk 2]
```diff
@@ -42,6 +46,101 @@ var gitIntegration = git.Integration{
...
+				return fmt.Errorf("error deserializing commit %s for user mapping: %v", commit.Sha, err)
+			}
+			gitName, gitEmail := parseRaw(c.Author.Raw)
+			var bitbucketData bool
+			bitbucketRefID := ""
+			if c.Author.User != nil && c.Author.User.AccountID != nil {
+				bitbucketData = true
+				bitbucketRefID = *c.Author.User.AccountID
+			}
+			gitRefID := hash.Values(gitEmail)
```
Comment (jhaynie) [inline integrations/bitbucket/export.go]: all refid should be hash with customerID

Comment (Robindiddams) [reply integrations/bitbucket/export.go]: will do

**Pull request merged**